mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvpManager.java

```diff
@@ -247,6 +252,54 @@ public boolean deleteKitByName(Transaction transaction, String kitName) {
 		return true;
 	}
 
+	/**
+	 * Begins listing bounties. Gives the first page, from which it is possible
+	 * to navigate to further pages
+	 *
+	 * @param tx the transaction
+	 * @param countPerPage the amount of bounties on each page
+	 * @return the first page of bounties, or an empty optional if there are no pages
+	 * @throws IllegalArgumentException if {@code countPerPage} is not positive
+	 */
+	public Optional<BountyPage> listBounties(Transaction tx, int countPerPage) {
+		if (countPerPage <= 0) {
+			throw new IllegalArgumentException("Count per page must be positive");
+		}
+		return listBounties(tx, countPerPage, null);
+	}
+
+	Optional<BountyPage> listBounties(Transaction tx, int countPerPage, BigDecimal afterAmount) {
+		DSLContext context = tx.getProperty(DSLContext.class);
+		var step = context
+				.select(LATEST_NAMES.USERNAME, KITPVP_STATISTICS.BOUNTY)
+				.from(KITPVP_STATISTICS)
+				.innerJoin(LATEST_NAMES)
+				.on(KITPVP_STATISTICS.USER_ID.eq(LATEST_NAMES.USER_ID));
+		RecordMapper<Record2<String, BigDecimal>, Bounty> mapper =
+				(record) -> new Bounty(record.value1(), record.value2());
+		List<Bounty> bounties;
+		if (afterAmount == null) {
```

Comment:
afterAmount is always null, since listBounties always calls it with null?

Reply:
The BountyPage implementation of `nextPage` calls listBounties with a non-null value.

For context, I'm using keyset-based pagination, by using the last bounty amount as the cursor. See also https://use-the-index-luke.com/no-offset